backend/tomato/lib/schema.py

```diff
@@ -125,14 +125,14 @@ def __init__(self, strict=False, **kwargs):
 		minLength = kwargs.pop('minLength', 3)
 		maxLength = kwargs.pop('maxLength', 100)
 		if strict:
-			regex = kwargs.pop('regex', "[a-zA-Z_][a-zA-Z0-9_]*")
+			regex = kwargs.pop('regex', "^[a-zA-Z_][a-zA-Z0-9_]*$")
```

Comment:
Please change the usage of regex in `String.check` to `re.match("^%s$" % self.regex, value)` instead of changing all regexs